lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -836,10 +900,20 @@ defmodule LightningWeb.WorkflowLive.Edit do
 
   @impl true
   def mount(_params, _session, %{assigns: assigns} = socket) do
+    pid = self()
+
+    view_only_users_ids =
+      Lightning.Repo.preload(assigns.project, project_users: [:user])
+      |> Map.get(:project_users)
+      |> Enum.filter(fn pu -> pu.role == :viewer end)
+      |> Enum.map(fn pu -> pu.user.id end)
+
     {:ok,
      socket
      |> authorize()
      |> assign(
+       pid: pid,
```

Comment:
Not sure we need to keep the `pid` in assigns right? Why not call `self()` in the functions?

Reply:
Fixed